.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,18 @@
+
+# One line description
+
+Please include a paragraph here describing what this PR does:
+what kind of change (refactoring, bugfix, feature), what is changed
+(short summary of new features or fix).
+
+If bugs are addressed in this PR, include the issue numbers.
+
+# Who benefits?
+
+Axom team, "LLNL customers" if you can't mention in detail,
+specific LLNL projects if you can, or outside feature request.
+
+# Design review
```

Comment:
What is the intent of this section? 

I think this adds too much overhead in our workflow/process with little or no value. The goal of the templates should be to help us be efficient, but, if we have to write a dissertation for each PR, I am afraid this would slow things down. 

Instead the high-level design should be succinctly captured in the Doxygen documentation and reflected by the Git history of the corresponding branch.

Reply:
@gzagaris The intent is to have an initial presentation of goals and approach, to get early feedback from the rest of the development team.  This will allow adjustment to be made earlier in the process, so we have more understanding up front and less monolithic PRs.  With monolithic PRs, we get big chunks of work -- but there are too many details buried in code, it's hard to decouple the code changes with why they are being made. (The same can actually apply to docs as well -- overall context isn't there yet)

This isn't replacement for full user or developer docs, but instead a more nimble and timely presentation of salient pieces (goals and approach) that help provide context for the rest of the Axom development team.

Reply:
@cyrush, what you describe sounds reasonable. I would include this information in the paragraph description of the PR. It is not clear to me that we need a `Design review` section. 

Perhaps, It would be more helpful to ask specific questions instead of leaving it open-ended and subject to misinterpretation about what should be included in each section. For example:

- What are the goal(s) of this PR?
- State the high-level technical approach to achieve the goals of this PR

Reply:
My intent for this section and *(brief)* paragraph was to encourage us to have early (not late) discussions of large code additions or API changes.  @gzagaris , you're right: pointless workflow overhead is to be avoided, and for small PRs we don't need this section. 

For large PRs, I strongly feel that the author should early on get agreement from colleagues on design principles.  If the PR has this specific section, it shows that the author took care to do this and reminds colleagues of their agreement.  This is why I made it a separate section.

@gzagaris , that's a good point about specific questions instead of open-ended prompts.  I'll make the change.

Reply:
Reformatted into a list with (areas) to fill in.